integration_test/src/main.rs

```diff
@@ -778,6 +786,97 @@ fn test_rescan_blockchain(cl: &Client) {
     assert_eq!(stop, Some(count - 1));
 }
 
+struct WalletParams<'a> {
+    name: &'a str,
+    disable_private_keys: Option<bool>,
+    blank: Option<bool>,
+    passphrase: Option<&'a str>,
+    avoid_reuse: Option<bool>,
+}
+
+fn build_wallet_params<'a>(
+    name: &'a str,
+    disable_private_keys: Option<bool>,
+    blank: Option<bool>,
+    passphrase: Option<&'a str>,
+    avoid_reuse: Option<bool>,
+) -> WalletParams<'a> {
+    WalletParams {
+        name,
+        disable_private_keys,
+        blank,
+        passphrase,
+        avoid_reuse,
+    }
+}
+
+fn test_create_single_wallet(cl: &Client, wallet_param: WalletParams) {
+    let result = cl
+        .create_wallet(
+            wallet_param.name,
+            wallet_param.disable_private_keys,
+            wallet_param.blank,
+            wallet_param.passphrase,
+            wallet_param.avoid_reuse,
+        )
+        .unwrap();
+
+    assert_eq!(result.name, wallet_param.name);
+    let expected_warning = match (wallet_param.passphrase, wallet_param.avoid_reuse) {
+        (None, Some(true)) => Some("Empty string given as passphrase, wallet will not be encrypted.".to_string()),
+        _ => Some("".to_string()),
+    };
+    assert_eq!(result.warning, expected_warning);
+
+    let wallet_client_url = format!("{}{}{}", get_rpc_url(), "/wallet/", wallet_param.name);
+    let wallet_client = Client::new(wallet_client_url, get_auth()).unwrap();
+    let wallet_info = wallet_client.get_wallet_info().unwrap();
+
+    assert_eq!(wallet_info.wallet_name, wallet_param.name);
+
+    fn option_to_bool(option: Option<bool>) -> bool {
+        match option {
+            None | Some(false) => false,
+            Some(true) => true,
+        }
+    }
+
+    let has_private_keys = !option_to_bool(wallet_param.disable_private_keys);
+    assert_eq!(wallet_info.private_keys_enabled, has_private_keys);
+    let has_hd_seed = has_private_keys && !option_to_bool(wallet_param.blank);
+    assert_eq!(wallet_info.hd_seed_id.is_some(), has_hd_seed);
+    let has_avoid_reuse = option_to_bool(wallet_param.avoid_reuse);
+    assert_eq!(option_to_bool(wallet_info.avoid_reuse), has_avoid_reuse);
+}
+
+fn test_create_wallet(cl: &Client) {
+    let wallet_names = vec!["alice", "bob", "carol", "denise", "emily"];
+    let mut wallet_params = vec![
+        build_wallet_params(wallet_names[0], None, None, None, None),
+        build_wallet_params(wallet_names[1], Some(true), None, None, None),
+        build_wallet_params(wallet_names[2], None, Some(true), None, None),
+    ];
+
+    if version() >= 190000 {
+        wallet_params.push(build_wallet_params(wallet_names[3], None, None, Some("pass"), None));
+        wallet_params.push(build_wallet_params(wallet_names[4], None, None, None, Some(true)));
+    }
+
+    for param in wallet_params {
+        test_create_single_wallet(&cl, param);
```

Comment:
This method is only used here, right? I wouldn't mind just having it expanded inside this for loop. Than the `WalletParams` type can also be made method-local. It's just a suggestion won't be blocking for this.

Reply:
Expanded in loop and put `WalletParams` as method-local.